src/gentropy/datasource/biosample_ontologies/utils.py

```diff
@@ -0,0 +1,197 @@
+"""Utility functions for Biosample ontology processing."""
+from functools import reduce
+
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.functions import (
+    array_distinct,
+    coalesce,
+    col,
+    collect_list,
+    collect_set,
+    explode_outer,
+    first,
+    regexp_replace,
+    udf,
+)
+from pyspark.sql.types import ArrayType, StringType
```

Comment:
This is just stylistic comment, but we do not import spark functions like this. There's a not too serious explanation for this: some of the functions might overwrite Python base function eg. `from pyspark.sql.functions import sum, max`, that's why we import all functions with a prefix: 

```python
import pyspark.sql.functions as f

# then calling a function would look like this:
f.array_distinct(f.array_union(f.col('colname')....
```
It's quite marginal, but being stylistically consistent is good.